src/util/memory.rs

```diff
@@ -147,6 +147,7 @@ pub fn handle_mmap_error<VM: VMBinding>(error: Error, tls: VMThread) -> ! {
     match error.kind() {
         // From Rust nightly 2021-05-12, we started to see Rust added this ErrorKind.
         ErrorKind::OutOfMemory => {
+            eprintln!("{}", get_process_memory_maps());
```

Comment:
Well, I just realised that the out-of-memory event is not always lethal because the VM binding can implement `VM::VMCollection::out_of_memory`.  If the VM can recover from OOM (like OpenJDK), it is probably unnecessary to print the memory maps.  But if it crashes (like when OpenJDK generates something like `hs_err_pidXXXXX.log`), the ability to print mmap would be helpful.  I think if there is a call to `VM::VMCollection::out_of_memory` following this, we should move it to the default implementation of `Collection::out_of_memory` (because it panics anyway).

Reply:
No this is different. This is an `mmap` error, not a simple OOM error. The simple OOM error is `AllocationError::HeapOutOfMemory`